tests/rest/admin/test_room.py

```diff
@@ -941,6 +941,52 @@ def _search_test(
         _search_test(None, "bar")
         _search_test(None, "", expected_http_code=400)
 
+    def test_search_term_non_ASCII(self):
+        """Test that searching for a room with non-ASCII characters works correctly"""
+
+        # Create test room
+        room_id = self.helper.create_room_as(self.admin_user, tok=self.admin_user_tok)
+
+        room_name = "ж"
+
+        # Set the name for the room
+        self.helper.send_state(
+            room_id,
+            "m.room.name",
+            {"name": room_name},
+            tok=self.admin_user_tok,
+        )
+
+        def _search_test_utf8(
```

Comment:
Since we're only calling this once, I don't think there's much value in it being an inner-function. It hink we can just inline the code.